svm/src/transaction_processor.rs

```diff
@@ -599,13 +602,16 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         } = checked_details;
 
         let fee_budget_limits = FeeBudgetLimits::from(compute_budget_limits);
-        let fee_details = solana_fee::calculate_fee_details(
-            message,
-            lamports_per_signature == 0,
-            fee_lamports_per_signature,
-            fee_budget_limits.prioritization_fee,
-            FeeFeatures::from(account_loader.feature_set.as_ref()),
-        );
+        let fee_details = if lamports_per_signature == 0 {
+            FeeDetails::default()
```

Comment:
@pgarg66 I missed this additional if when I reviewed. Why is this necessary?

Reply:
We want to remove this behavior from fee calculations, adding the special-casing (test-only) in another spot complicates that.

Reply:
Adding of special-casing in an interface seemed like a wrong approach. As other (non-agave) users of SVM should not care about supporting this flag. We can remove this code from all the places altogether. The interface won't require a change with this approach.

Reply:
Ah I see now. Yeah would prefer to stabilize the interface as is - and we can (should) remove this special-casing as soon as possible.

I have had a draft PR to do it, but have had trouble since SO many tests break if it's removed. Haven't had a chance to pick it back up recently.

Reply:
Yes, I noticed your previous PR that you backed out. It'll be great to remove this test-only flag.